openeo/UDP.ipynb

```diff
@@ -0,0 +1,278 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# User-Defined Processes(UDP)\n",
+    "\n",
+    "openEO API allows processes to be chained together in a process graph to build a particular algorithm. Often, users have specific (sub)chains that reoccur in the same process graph or even in different process graphs or algorithms. Here user can store such (sub)chains on the back-end as a so-called user-defined process. This allows you to build your library of reusable building blocks. Ultimately, the openEO API allows you to publicly expose your user-defined process, so that other users can invoke it as a service.\n",
+    "\n",
+    "This notebook provides a step-by-step guide on how to create your User-Defined Process (UDP) for Normalized Difference Water Index (NDWI) analysis. The guide covers the fundamental steps that need to be followed to create the UDP."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import necessary packages\n",
+    "\n",
+    "from openeo.processes import array_create\n",
+    "from openeo.api.process import Parameter\n",
+    "import openeo"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Authenticated using refresh token.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# establish a connection to the backend and authenticate\n",
+    "\n",
+    "connection = openeo.connect(url = \"openeo-staging.creo.vito.be\").authenticate_oidc()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The openEO Python client lets you define parameters as `Parameter` instances. In general you have to specify at least the parameter name, a description and a schema."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define input parameters\n",
+    "time_param = Parameter(\n",
+    "                            name = 'date',\n",
+    "                            description = \"A date within wich you wish to load the Terrasope NDVI product.\",\n",
+    "                            schema = {\"type\":\"array\",\"subtype\":\"data-time\"}\n",
```

Comment:
I'm confused here: `time_param` looks like  single date so must be a string I think, but here it's `"type":"array"`. 

Also `"subtype":"data-time"` look invalid (e.g. `data-time` is not a thing)

Reply:
> Another thing: I find this indentation approach very ugly and wasteful
> The conventional way is to just indent with 4 spaces

Updated

Reply:
> I'm confused here: `time_param` looks like single date so must be a string I think, but here it's `"type":"array"`.
> 
> Also `"subtype":"data-time"` look invalid (e.g. `data-time` is not a thing)

Updated as string